### PR TITLE
Log amount of found clusters in kmeans init

### PR DESCRIPTION
**Pull request by JulianWgs**
- Put the changes behind the old information to not destroy old processing pipelines (if there are any)

### CODE REVIEW

File: dask_ml/cluster/k_means.py [hunk 1]
```diff
@@ -447,7 +447,10 @@ def init_scalable(
     # to do that.
 
     if len(centers) < n_clusters:
-        logger.warning("Found fewer than %d clusters in init.", n_clusters)
+        logger.warning(
```
Comment (TomAugspurger) [inline dask_ml/cluster/k_means.py:450]: Can we keep the `%d` style formatting? It avoids the cost of formatting when the logger isn't enabled.

Comment (JulianWgs) [reply dask_ml/cluster/k_means.py:450]: You mean that the f-string syntax is slower?

Comment (TomAugspurger) [reply dask_ml/cluster/k_means.py:450]: Can be, so I prefer to consistently use `%`-style for logging.

**Comment by TomAugspurger** [comment]
Thanks. The sklearn dev CI issue is unrelated. I'll debug that separately.

**Pull request merged**